tests/test_rpc.py

```diff
@@ -112,6 +112,25 @@ def long_call(self, close_proxy, return_value=None):
         return return_value
 
 
+class CaptureErrorServer(RPCServerThread):
+    """
+    This is the RPCServerThread but it captures any errors
```

Comment:
Ah, I like it! Smart.

Reply:
yeah I like it as well but as per our discussion I'll remove it, as we are not going to go and fix the issues it is raising right now in the tests.

It did actually however prove that my fix worked.

Reply:
OK I've actually left it in and commented out the raising. How do you feel about that?
Would you prefer me to remove it entirely. I feel by leaving it in there we are drawing attention to the fact that we are not capturing the errors from the server and providing a solution that could be enabled when we are ready to deal with the errors.

another approach I guess would be to remove the class but just leave a TODO with the link to the SO answer that gave me the solution.

Reply:
Well, if you have inspired the code from something you saw on StackOverflow, you are legally required to include a link. All posts on StackOverflow are done under the [Creative Commons](https://creativecommons.org/licenses/by-sa/4.0/) license. It requires that any piece of code you reuse or adapt is given attribution by providing a link to the required post.

I also just learned that you need to mention if the code was modified or as is. I've generally been using the following formulations: "This code has been lifted" vs "This code has been inspired by".

Reply:
I'd rather put a comment in the code itself with a FIXME saying that a excetion can be raised from here, so the next person who needs to fix this can do it can find the place to fix right away.

Reply:
AH OK I didn't know about the inspired by bit, but I did have the link in there.

I have a TODO which I could rename to FIXME, are you saying just have a comment/FIXME with the link and remove the class entirely rather than just commenting out the raise bit in the class?

Reply:
From my understand the class is not used. Instead of leaving a test class with some magic code in it, let's document the actual class with its shortcoming. Does that make sense?